### PR TITLE
gh-81548: Clarify the deprecation of octal sequences affect byte strings

### PR DESCRIPTION
**Pull request by pablogsal**
Automerge-Triggered-By: GH:pablogsal

### CODE REVIEW

File: Doc/whatsnew/3.11.rst [hunk 1]
```diff
@@ -1100,8 +1100,8 @@ Deprecated
   that was added in Python 3.10.
   (Contributed by Raymond Hettinger in :gh:`89519`.)
 
-* Octal escapes with value larger than ``0o377`` now produce
-  a :exc:`DeprecationWarning`.
+* Octal escapes in :class:`bytes` objects with value larger than ``0o377`` now
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.11.rst]: in string and bytes literals

Comment (serhiy-storchaka) [inline Doc/whatsnew/3.11.rst]:
```suggestion
* Octal escapes in string and bytes literals with value larger than ``0o377`` now
```

**Review by serhiy-storchaka — APPROVED** [review]

File: Doc/whatsnew/3.11.rst [hunk 2]
```diff
@@ -1100,8 +1100,8 @@ Deprecated
   that was added in Python 3.10.
   (Contributed by Raymond Hettinger in :gh:`89519`.)
 
-* Octal escapes with value larger than ``0o377`` now produce
-  a :exc:`DeprecationWarning`.
+* Octal escapes in string and bytes literals with value larger than ``0o377`` now
+  produce a :exc:`DeprecationWarning`.
```
Comment (slateny) [inline Doc/whatsnew/3.11.rst]: I think 'a' could be optionally left out: '...now produce DeprecationWarning'

Comment (pablogsal) [inline Doc/whatsnew/3.11.rst]:
```suggestion
  produce :exc:`DeprecationWarning`.
```

**Pull request merged**